apps/dashboard/src/components/primitives/table.tsx

```diff
@@ -46,62 +57,80 @@ const Table = React.forwardRef<HTMLTableElement, TableProps>(
 );
 Table.displayName = 'Table';
 
-const TableHeader = React.forwardRef<HTMLTableSectionElement, React.HTMLAttributes<HTMLTableSectionElement>>(
-  ({ className, ...props }, ref) => (
-    <thead
-      ref={ref}
-      className={cn('sticky top-0 bg-neutral-50 shadow-[0_0_0_1px_hsl(var(--neutral-alpha-200))]', className)}
-      {...props}
-    />
-  )
-);
+const TableHeader = React.forwardRef<HTMLTableSectionElement, TableHeaderProps>(({ className, ...props }, ref) => (
+  <thead
+    ref={ref}
+    className={cn('sticky top-0 bg-neutral-50 shadow-[0_0_0_1px_hsl(var(--neutral-alpha-200))]', className)}
+    {...props}
+  />
+));
 TableHeader.displayName = 'TableHeader';
 
-const TableBody = React.forwardRef<HTMLTableSectionElement, React.HTMLAttributes<HTMLTableSectionElement>>(
-  ({ className, ...props }, ref) => <tbody ref={ref} className={cn('', className)} {...props} />
+const TableHead = React.forwardRef<HTMLTableCellElement, TableHeadProps>(
+  ({ className, children, sortable, sortDirection, onSort, ...props }, ref) => {
+    const content = (
+      <div className={cn('flex items-center gap-1', sortable && 'hover:text-foreground-900 cursor-pointer')}>
+        {children}
+        {sortable && (
+          <>
+            {sortDirection === 'asc' && <RiArrowUpSFill className="text-text-sub-600 size-4" />}
+            {sortDirection === 'desc' && <RiArrowDownSFill className="text-text-sub-600 size-4" />}
+            {!sortDirection && <RiExpandUpDownFill className="text-text-sub-600 size-4" />}
+          </>
+        )}
+      </div>
+    );
+
+    return (
+      <th
+        ref={ref}
+        className={cn(
+          'text-foreground-600 h-10 px-6 py-2 text-left align-middle font-medium [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]',
```

Comment:
why don't we use then `role="checkbox"` ?

Reply:
This is for future proofing taken from align UI, in case we will want to add a checkbox column for other tables. Just made sure they are aligned in implementation, not necessary needed for this particular PR